test_scripts/Polices/External_UCS_Informing_HMI/020_ATF_N_GetListofPermissions_empty_due_to_invalid_param_entityID_OnAppPermissionChange.lua

```diff
@@ -0,0 +1,142 @@
+-- [HMI API] ExternalConsentStatus struct & EntityStatus enum
+--
+-- Description:
+-- For Genivi applicable ONLY for 'EXTERNAL_PROPRIETARY' Polcies
+-- Check that SDL invalidates notification OnAppPermissionConsent due to invalid value type of parameter entityID
+--
+-- 1. Used preconditions
+-- SDL is built with External_Proprietary flag
+-- SDL and HMI are running
+-- Application is registered and activated
+-- PTU file is updated and application is assigned to functional groups: Base-4, user-consent groups: Location-1 and Notifications
+-- PTU has passed successfully
+-- HMI sends <externalConsentStatus> to SDl via OnAppPermissionConsent (parameter entityID has invalid value type, rest of params present and within bounds, EntityStatus = 'ON')
+-- SDL doesn't receive updated Permission items and consent status
+--
+-- 2. Performed steps
+-- HMI sends to SDL GetListOfPermissions (appID)
+--
+-- Expected result:
+-- SDL sends to HMI empty array
+---------------------------------------------------------------------------------------------
+
+--[[ General configuration parameters ]]
+config.deviceMAC = "12ca17b49af2289436f303e0166030a21e525d266e209267433801a8fd4071a0"
+-- ToDo (vvvakulenko): remove after issue "ATF does not stop HB timers by closing session and connection" is resolved
+config.defaultProtocolVersion = 2
+
+--[[ Required Shared libraries ]]
+local commonFunctions = require ('user_modules/shared_testcases/commonFunctions')
+local commonSteps = require('user_modules/shared_testcases/commonSteps')
+local testCasesForPolicyTable = require('user_modules/shared_testcases/testCasesForPolicyTable')
+local commonTestCases = require('user_modules/shared_testcases/commonTestCases')
+
+--[[Local Variables]]
+local params_invalid_data =
+{
+  {param_value = "invalidValue", comment = "String"}
+  {param_value = 1.32 comment = "Float"},
+  {param_value = {}, comment = "Empty table" },
+  {param_value = { entityType = 1, entityID = 1 }, comment = "Non-empty table"},
+  {param_value = -1, comment = "OutOfLowerBound"},
+  {param_value = 130, comment = "OutOfUpperBound" },
+  {param_value = "", comment = "Empty" },
+  {param_value = nil, desc = "Null" }
+}
+
+--[[ General Precondition before ATF start ]]
+commonFunctions:SDLForceStop()
+commonSteps:DeleteLogsFiles()
+commonSteps:DeletePolicyTable()
+
+--[[ General Settings for configuration ]]
+Test = require('connecttest')
+require('cardinalities')
+require('user_modules/AppTypes')
+
+--[[ Preconditions ]]
+commonFunctions:newTestCasesGroup("Preconditions")
+
+function Test:Precondition_trigger_getting_device_consent()
+  testCasesForPolicyTable:trigger_getting_device_consent(self, config.application1.registerAppInterfaceParams.appName, config.deviceMAC)
+end
+
+for i = 1, #params_invalid_data do
+  Test["Precondition_PTU_and_OnAppPermissionConsent_Invalid_"..params_invalid_data[i]] = function(self)
+    local ptu_file_path = "files/jsons/Policies/Related_HMI_API/"
```

Comment:
@vvvakulenko, To have correct PTU and OnAppPermissionChange notification sent after each PTU, you should change permission in ptu file. So you need as example:

local ptu_file
local consentedgroups
local allowed_func

if ( (i % 2) == 0 ) then
ptu_file = "OnAppPermissionConsent_ptu1.json"
consentedgroups = {{ allowed = true, id = 4734356, name = "DrivingCharacteristics-3"}}
allowed_func = { { name = "DrivingCharacteristics", id = 4734356}}
else
ptu_file = "OnAppPermissionConsent_ptu.json"
consentedgroups = {
{ allowed = true, id = 156072572, name = "Location-1"},
{ allowed = true, id = 1809526495, name = "Notifications"}
}

  allowed_func = {
    { name = "Location", id = 156072572},
    { name = "Notifications", id = 1809526495}
  }

end